com.unity.perception/Runtime/Randomization/Randomizers/RandomizerExamples/Utilities/PoissonDiskSampling.cs

```diff
@@ -71,6 +85,44 @@ public void Execute()
             }
         }
 
+        /// <summary>
+        /// This job is for filtering out all super sampled Poisson points that are found outside of the originally
+        /// specified 2D region. This job will also shift the cropped points back to their original region.
+        /// </summary>
+        [BurstCompile]
+        struct CropJob : IJob
+        {
+            public float width;
+            public float height;
+            public float minimumRadius;
+            [ReadOnly] public NativeArray<float2> superSampledPoints;
+            public NativeList<float2> croppedSamples;
+
+            public void Execute()
+            {
+                var results = new NativeArray<bool>(
+                    superSampledPoints.Length, Allocator.Temp, NativeArrayOptions.UninitializedMemory);
+
+                for (var i = 0; i < superSampledPoints.Length; i++)
+                {
+                    var point = superSampledPoints[i];
+                    results[i] = point.x >= minimumRadius && point.x <= width + minimumRadius
+                        && point.y >= minimumRadius && point.y <= height + minimumRadius;
+                }
+
+                for (var i = 0; i < superSampledPoints.Length; i++)
+                {
+                    if (results[i])
+                    {
+                        var p = superSampledPoints[i];
+                        croppedSamples.Add(new float2(p.x - minimumRadius, p.y - minimumRadius));
+                    }
```

Comment:
I don't understand why you are looping through this data twice. Couldn't this just be:

` for (var i = 0; i < superSampledPoints.Length; i++)
 {
     var point = superSampledPoints[i];
     if (point.x >= minimumRadius && point.x <= width + minimumRadius
           && point.y >= minimumRadius && point.y <= height + minimumRadius)
     {
         croppedSamples.Add(new float2(p.x - minimumRadius, p.y - minimumRadius));
     }          
}`


If you did that you don't have to allocated the results array.

Reply:
Ahh, so I separated this code into two pieces so that burst could vectorize the first loop and do less work on the un-vectorizable second loop. I'll annotate the code with this information.

Reply:
Technically, there's a multi-threaded way of splitting off this filtering work with the [IJobParallelForFilter](https://docs.unity3d.com/Packages/com.unity.jobs@0.0/api/Unity.Jobs.IJobParallelForFilter.html) interface, but I thought that was overkill.

Reply:
Here's a neat look at the burst inspector to see all the vectorization it does at the first loop (basically every purple instruction):
![image](https://user-images.githubusercontent.com/53197518/113598033-663f1800-95f1-11eb-9c41-81e79715c422.png)

Reply:
In contrast, burst can't seem to vectorize any part of the second loop:
![image](https://user-images.githubusercontent.com/53197518/113598346-d9488e80-95f1-11eb-9272-e17019d87df1.png)

Reply:
Interesting. I figured you had a good reason for it